### PR TITLE
[WIP] Berlin event page

### PR DESCRIPTION
**Pull request by aulneau**
Here's the new page for the Berlin event! I know various people expressed interest in keeping this work minimal, so I just spent today on it and this is where I am. 

Things I'm still waiting on / need to finish: 
- Videos
- Review of organization
- Review of copy in hero (cc @pstan26 @xanbots ?)
- Need to add arrows for the slider
- Would like the side panel sections to auto expand based off of scroll position

To do:
- [x] Update talks to new titles
- [x] Add in / update photos in slider
- [x] Add scroll position awareness to sidebar
- [x] Add in video links (once I get them)
- [x] Add in press logos / links
- [x] Add learning sessions 

Structural changes / features: 
- Added SCSS support
- Updated a few more deps (browserify, babelify, etc)
- Added React-helmet for better SEO once we implement something like prerender.io (see https://github.com/blockstack/blockstack.org/issues/532)
- Images on this page are being run through imgix.com (an amazing service, automatically crops to faces, will render webp if the browser supports it, automatic compression, etc)
- Each talk is linkable, eg `#talk-title` along with each section, eg `#morning-sessions`

See preview here -> https://deploy-preview-533--blockstack.netlify.com/berlin2018

cc @jeffdomke @shea256

### CODE REVIEW

**Comment by yknl** [comment]
Looks great @aulneau!

One thing is that on a mobile screen size, the video screenshots alternates above and below the title of the video. This might be a bit confusing, maybe we should have them be consistent on mobile?

image

**Comment by pstan26** [comment]
looking nice!

Will have videos in unlisted Youtube format ready by Friday. @aulneau how will they play when you click them? pop out or new page or what?

Would change the copy to:
The Blockstack Signature Fund and the folks who brought you Blockstack Summit 2017 are excited *to bring you* Blockstack Berlin, the first in a series of worldwide events and demo days around Blockstack. The event took place on March 2, 2018 in Berlin at the Axica Convention Center.

File: app/js/pages/Berlin2018Page.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    title: 'After Lunch Sessions',
+    items: filterIndex(mainEvent, 9, 11),
+  },
+  {
+    title: 'App Demos',
+    items: filterIndex(mainEvent, 12, 23),
+  },
+  {
+    title: 'Late Afternoon Sessions',
+    items: filterIndex(mainEvent, 24, 26),
```
Comment (cwackerfuss) [inline app/js/pages/Berlin2018Page.js]: I like how thoughtful this is laid out codewise. The `filterIndex` idea feels a bit brittle to me. Can we quickly switch this to filtering on a new key that we add to events? For instance:
```
items: mainEvent.filter(item => item.type === 'lateAfternoonSession')
```

File: app/js/pages/Berlin2018Page.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+    if (min && !max) {
+      return i >= min
+    }
+    if (!min && max) {
+      return i <= max
+    }
+  })
+
+const filteredData = [...removeDuplicates(berlinData, 'talkTitle')]
```
Comment (cwackerfuss) [inline app/js/pages/Berlin2018Page.js]: We manage the data here -- is this logic necessary given that we could just remove any duplicate talk titles in the static data?

File: app/js/pages/Berlin2018Page.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+import React, { Component, Fragment } from 'react'
+import { Helmet } from 'react-helmet'
+import CollapsibleList from '../components/CollapsibleList'
+import berlinData from '../datastore/berlin-event-data'
+import Slider from 'react-slick'
+import SignatureTalks from '../components/SignatureTalks'
+import { slugify } from '../utils/functions'
+
+const settings = {
```
Comment (cwackerfuss) [inline app/js/pages/Berlin2018Page.js]: Lets call this `sliderSettings`

File: app/js/pages/Berlin2018Page.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  },
+  {
+    src:
+      'https://blockstack.imgix.net/b0415647-dc2f-44ac-b083-e7f445229f10__3020103.jpg',
+  },
+  {
+    src:
+      'https://blockstack.imgix.net/2a992e79-07b5-4440-a681-6c33a339a5ad__3020104.jpg',
+  },
+]
```
Comment (cwackerfuss) [inline app/js/pages/Berlin2018Page.js]: Dern prettier, making the formatting weird here.

File: app/js/pages/Berlin2018Page.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    <div key={i} id={slugify(section.title)}>
+      <h2 className="event__talk__header">{section.title}</h2>
+      <SignatureTalks talks={section.items} allData={berlinData} />
+    </div>
+  ))
+
+class Berlin2018Page extends Component {
+  render() {
+    return (
+      <div className="page--berlin-event" id="berlin-page">
```
Comment (cwackerfuss) [inline app/js/pages/Berlin2018Page.js]: Love that you're using BEM, we just don't really use it in the project anywhere else. I'm not going to make a fuss about it or say you should change it, but there should be a consistent standard in style across the app (even if its not ideal, at least its consistent). Just something to consider for the future!

**Comment by aulneau** [comment]
@pstan26 Awesome, yeah either option will work. I saw that the previous pages we have just open up the video into youtube, but I can make it popup in the page itself. I'd prefer them to stay in the page, so I'll spend some time tomorrow building out a little modal for the videos. Do you have a preference.

good catch @yknl, should be fixed now! :)

**Comment by xanbots** [comment]
Also @aulneau @pstan26 what about putting the Blockstack Berlin logo at the top, centered?

File: gulp/tasks/copyStyles.js [hunk 6]
```diff
@@ -3,12 +3,14 @@
...
   return gulp.src(
     [
-      config.sourceDir + 'styles/*.css',
       config.sourceDir + '../node_modules/bootstrap/dist/css/*',
       config.sourceDir + '../node_modules/font-awesome/css/*',
+      config.sourceDir + '../node_modules/slick-carousel/slick/slick.css',
+      config.sourceDir + '../node_modules/slick-carousel/slick/slick-theme.css',
       config.sourceDir + '../node_modules/highlight.js/src/styles/sunburst.css'
     ]
   ).pipe(gulp.dest(config.buildDir + 'css/'));
```
Comment (cwackerfuss) [inline gulp/tasks/copyStyles.js]: Wonder if there's an opportunity here to just concat all the css files.

File: package.json [hunk 7]
```diff
@@ -113,5 +56,65 @@
...
+    "build": "npm run fetch-docs; gulp prod",
+    "build-docs": "npm run fetch-docs; gulp buildDocs",
+    "dev": "npm run fetch-docs; gulp dev",
+    "fetch-docs": "git submodule init; git submodule update --remote",
+    "format": "prettier --write \"app/**/*.js\" && echo -e '\\033[0;32m'💅 Just use prettier'\\033[0m'",
+    "postinstall": "cd ./node_modules/highlight.js && npm i && node ./tools/build.js :common",
+    "prod": "npm run fetch-docs; gulp prod",
+    "start": "serve ./build --single"
   }
 }
```
Comment (cwackerfuss) [inline package.json]: Looks like the package.json modifier crept in again 🕺

File: app/js/pages/Berlin2018Page.js (hunk 2 above)
Comment (aulneau) [reply app/js/pages/Berlin2018Page.js]: The duplicates are defining talk <-> speaker relationships. I don't think we should remove the duplicates from the original data.

**Comment by jeffdomke** [comment]
Looking fresh. Some design suggestions: 
- Feel like the video thumbs are pretty big
- Video thumbs should be clickable with a clear affordance as play button
- Change above might make the "watch" button redundant. 
- Times can be de-emphasized. Removed? 
- Active state on the nav items as I scroll? 
- Bigger thought: Is the nav and the content semi-redundant forms of the same info? If I'm hunting through the nav for a certain speaker opposed to topic, can I find it? Is the full list of content really that onerous as to need a nav? The answer might be yes, I'm just thinking out loud. 

If I missed the boat on feedback when this was a static mock then apologies. 😞

**Comment by aulneau** [comment]
Awesome! thanks for the feedback @jeffdomke!

> Feel like the video thumbs are pretty big

Was waiting to have videos to see if they would work to just be embedded directly there, and will play with it more once I have a video in there. I was also thinking about selecting a photo from the photographer as a thumbnail dependent on the videos initial image. I'll play around with reducing their size a bit. I think when they are all different, the size might not be so overwhelming. 

> Video thumbs should be clickable with a clear affordance as play button
> Change above might make the "watch" button redundant.

Was pulling from this page: https://blockstack.org/videos I think it makes sense for both to be clickable, as to give people every opportunity to watch the video. I will add a hover state for the video if we do not go with embedding the youtube video directly here.

> Times can be de-emphasized. Removed?

Sure thing

> Active state on the nav items as I scroll?

Yep! See my list of things I'm adding.

> Bigger thought: Is the nav and the content semi-redundant forms of the same info? If I'm hunting through the nav for a certain speaker opposed to topic, can I find it? Is the full list of content really that onerous as to need a nav? The answer might be yes, I'm just thinking out loud.

I thought about that too. I opted for adding it because of the sheer number of items. If we also add in the learning session talks here, that's a pretty steep list. It'd be hard to find out what video I'd potentially like to watch from an initial glance at the page. I think the advantage of having the side nav outweighs any possible redundancy that it might create. I can add in the speakers too, the one talk with 5 participants just throws everything out of sorts, though.

> If I missed the boat on feedback when this was a static mock then apologies. 😞

Just jumped into coding this up! So you didn't miss any boat :)

**Comment by aulneau** [comment]
@cwackerfuss: Re: imgix, @GinaAbrams set the account up, so she is the key master! We can proxy any number of buckets for use with imgix.

Currently the imgix is proxied over an s3 bucket that was created when I signed up for a blockstack prismic.io account (no cost, under to@blockstack.org). They generate an s3 bucket for you to use with generous limits. I did that with the intention of testing out prismic for blockstack, but if ultimately we do not go with prismic, we can upload the images to another s3 bucket or similar storage solution and add that to the imgix proxy. 

Regarding the argument that we have not been using imgix for anything else, I think that what we are currently doing (including all photographic assets with the project and hosting on firebase) is not the best choice we could be making related to the mangement of our media assets for the site. I think moving forward we can start to use a dedicated bucket for media and use the imgix for processing, the benefits are just so great! automatic compression, cdn caching, on-the-fly manipulation/cropping, the list goes on!

**Comment by pstan26** [comment]
@aulneau 

>  I can make it popup in the page itself. I'd prefer them to stay in the page, so I'll spend some time tomorrow building out a little modal for the videos. Do you have a preference.

Popup would be preferable, we want people to stay on the Blockstack site instead of navigate away to Youtube. Thanks!

File: app/js/datastore/berlin-event-data.js [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+    talkTitle: 'Subdomains',
+  },
+  {
+    name: 'Elizabeth Stark',
+    company: 'Lightning Labs',
+    jobTitle: 'Co-Founder and CEO',
+    twitter: '@starkness',
+    day: 'March 2',
+    time: '11:42 AM',
+    talkTitle: 'Building the Future of Layer 2',
```
Comment (larrysalibra) [inline app/js/datastore/berlin-event-data.js]: This was changed on the event day:

Elizabeth Stark interview with Larry Salibra.

Add me! :-D

**Comment by larrysalibra** [comment]
@cwackerfuss 
> I'm concerned about consistency with the rest of the project. These images are now hosted with imgix, but the rest of the app uses a different strategy. We're going to frankenstein this thing real fast if we casually implement new tech

I am too - I'm going to add this to today's engineering meeting agenda so that we can discuss.

File: app/js/datastore/berlin-event-data.js (hunk 8 above)
Comment (aulneau) [reply app/js/datastore/berlin-event-data.js]: Oop sorry! Will add! :)

**Comment by aulneau** [comment]
@cwackerfuss

> I'm concerned about consistency with the rest of the project. These images are now hosted with imgix, but the rest of the app uses a different strategy. We're going to frankenstein this thing real fast if we casually implement new tech

> I am too - I'm going to add this to today's engineering meeting agenda so that we can discuss.

Just to clarify, none of these images are _hosted_ with imgix. They can be hosted anywhere. I've made a imgix proxy for all of the current assets we have contained within the repo:

eg: https://blockstack.org/images/resources/browser-home-screen@2x.png -> https://blockstackorg.imgix.net/resources/browser-home-screen@2x.png

With a simple replace to the image component in the repo, all of the images can point to the new imgix proxy and benefit from it:

```
    if(src.startsWith('/images/')){
      src = src.replace('/images/', 'https://blockstackorg.imgix.net/')
    }
```

**Comment by aulneau** [comment]
Okay -- barring any other changes and when the videos are put in, this should be good to go!

**Pull request merged**